anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/source/data/initializer/throttlegate/RandomThrottleGate.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2018-2019 Expedia Group, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expedia.adaptivealerting.anomdetect.source.data.initializer.throttlegate;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Random;
+
+@Slf4j
+public class RandomThrottleGate implements ThrottleGate {
```

Comment:
Please write a unit test for this.
Something that calls isOpen() 1,000 times and asserts that it returns `true` around `1000 * throttleGateLikelihood` of the time (within an acceptable range) would be good.